src/test/scala/com/ubirch/services/IncidentListenerSpec.scala

```diff
@@ -117,9 +127,211 @@ class IncidentListenerSpec extends TestBase with EmbeddedKafka with StrictLoggin
       }
       eventlogCounter mustBe 2
       niomonCounter mustBe 2
+    }
+  }
+}
 
+class IncidentListenerFuncSpec extends FreeSpec with MustMatchers {
```

Comment:
Added some test code for the functions of IncidentListener

Reply:
great.